pyrep/pyrep.py

```diff
@@ -235,8 +235,11 @@ def merge_objects(self, objects: List[Shape]) -> Shape:
         :return: A single merged shape.
         """
         handles = [o.get_handle() for o in objects]
-        handle = sim.simGroupShapes(handles, merge=True)
-        return Shape(handle)
+        # FIXME: sim.simGroupShapes(merge=True) won't return correct handle,
+        # so we use name to find correct handle of the merged shape.
+        name = objects[-1].get_name()
+        sim.simGroupShapes(handles, merge=True)
+        return Shape(name)
```

Comment:
@stepjam I think this is a but in the C++ layer. `simGroupShapes(merge=True)` returns `1` in the test case, which is not `Shape` object.